src/shared/components/case-editor/services/cases.service.ts

```diff
@@ -104,31 +105,65 @@ export class CasesService {
    *
    * @param jsonResponse - {}
    */
-  private handleNestedDynamicListsInComplexTypes(jsonResponse) {
+  private handleNestedDynamicLists(jsonResponse) {
 
     if (jsonResponse.case_fields) {
       jsonResponse.case_fields.forEach(caseField => {
-        if (caseField.field_type && caseField.field_type.type === CasesService.SERVER_RESPONSE_FIELD_TYPE_COMPLEX) {
-
-          caseField.field_type.complex_fields.forEach(field => {
-
-            if (field.field_type.type === CasesService.SERVER_RESPONSE_FIELD_TYPE_DYNAMIC_LIST) {
-              const list_items = caseField.value[field.id].list_items;
-              const value = caseField.value[field.id].value;
-              field.value = {
-                list_items: list_items,
-                value: value ? value : undefined
-              };
-              field.formatted_value = field.value;
-            }
-          });
+        if (caseField.field_type) {
+          this.setDynamicListDefinition(caseField, caseField.field_type, caseField);
         }
       });
     }
 
     return jsonResponse;
   }
 
+  private setDynamicListDefinition(caseField, caseFieldType, rootCaseField) {
+    if (caseFieldType.type === CasesService.SERVER_RESPONSE_FIELD_TYPE_COMPLEX) {
+
+      caseFieldType.complex_fields.forEach(field => {
+        try {
+          if (field.field_type.type === CasesService.SERVER_RESPONSE_FIELD_TYPE_DYNAMIC_LIST) {
+            const dynamicListValue = this.getDynamicListValue(rootCaseField.value, field.id);
+            const list_items = dynamicListValue.list_items;
+            const value = dynamicListValue.value;
+            field.value = {
+              list_items: list_items,
+              value: value ? value : undefined
+            };
+            field.formatted_value = {
+              ...field.formatted_value,
+              ...field.value
+            };
+          } else {
+            this.setDynamicListDefinition(field, field.field_type, rootCaseField);
+          }
+        } catch (error) {
+          console.log(error);
```

Comment:
Probably you could remove this console.log

Reply:
This handles the error of the try-catch, so I think we can leave it for now. As this is highly recursive code I want to be able to tell if the app runs into "Maximum call stack size exceeded" issues and rethink the solution in the future.